Day-7_Functions/functions.js

```diff
@@ -0,0 +1,601 @@
+        return true; 
+    } else if (Array.isArray(value) && value.length === 0) {
+        return true;
+    } 
+    return false; 
+}
+
+// Write a function called sumOfArrayItems, it takes an array parameter and return the sum of all the items. Check if all the array items are number types. If not give return reasonable feedback.
+function sumOfArrayItems(array) {
+    let sum = 0
+    for (let element of array) {
+        if (typeof element !== 'number') {
+            return "Error: All items in the array must be numbers."
+        }
+        sum += element
+    }
+    return sum
+}
+
+// Write a function called average, it takes an array parameter and returns the average of the items. Check if all the array items are number types. If not give return reasonable feedback.
+function average(array) {
+    let sum = 0;    
+
+    for (let element of array) {
+        if (typeof element !== 'number') {
+            return "Error: All items in the array must be numbers."
+        }
+        sum += element        
+    }
+    return sum / array.length
+}
+console.log(average([1, 2, 3, 4, 5]));
+
+// Write a function called modifyArray takes array as parameter and modifies the fifth item of the array and return the array. If the array length is less than five it return 'item not found'.
+// console.log(modifyArray(['Avocado', 'Tomato', 'Potato','Mango', 'Lemon','Carrot']);
+// ['Avocado', 'Tomato', 'Potato','Mango', 'LEMON', 'Carrot']
+// console.log(modifyArray(['Google', 'Facebook','Apple', 'Amazon','Microsoft',  'IBM']);
+// ['Google', 'Facebook','Apple', 'Amazon','MICROSOFT',  'IBM']
+// console.log(modifyArray(['Google', 'Facebook','Apple', 'Amazon']);
+//   'Not Found'
+function modifyArray(array) {    
+    if (array.length < 5) {
+        return 'Item not found'
+    }
+    
+    if (typeof array[4] === 'string') {
+        array[4] = array[4].toUpperCase();
+    }else{
+        return 'Fifth item is not a string'
+    }
+
+    return array
+}
+console.log(modifyArray(['Avocado', 'Tomato', 'Potato','Mango', 'Lemon','Carrot']))
+console.log(modifyArray(['Google', 'Facebook','Apple', 'Amazon','Microsoft',  'IBM']))
+console.log(modifyArray(['Google', 'Facebook','Apple', 'Amazon']))
+
+// Write a function called isPrime, which checks if a number is prime number.
+function isPrime(num) {
+    // Check if num is less than 2 or not an integer
+    if (num < 2 || num !== Math.round(num)) {
+        return false; // 1 and non-integers are not prime numbers
+    }    
+    for (let i = 2; i < num; i++) {
+        if (num % i === 0) {
+            return false
+        }
+    }
+    return true
+}
+console.log(isPrime(13));
+
+// Write a functions which checks if all items are unique in the array.
+function areAllUnique(array) {
+    const uniqueNumbers = []
+    for (let element of array) {
+        if (uniqueNumbers.includes(element)) {
+            return false
+        }
+        uniqueNumbers.push(element);
+    }
+    return true
```

Comment:
Can you do it without using for loop? Hint: use Set data structure.

Reply:
Updated